src/demos/datepicker/fuzzy-datepicker-demo.component.html

```diff
@@ -0,0 +1,133 @@
+      [minDate]="minDate"
+      [skyFuzzyDatepickerInput]
+      [skyDatepickerNoValidate]="noValidate"
+      [startingDay]="startingDay"
+      [yearRequired]="yearRequired"
+      [(ngModel)]="selectedDate"
+      #date="ngModel"
+    />
+  </sky-datepicker>
+
+  <p *ngIf="date.errors && date.errors.skyFuzzyDate && (date.dirty || date.touched)"
+    class="sky-error-label"
+  >
+    <span *ngIf="date.errors.skyFuzzyDate?.invalid">
+      Please enter a valid date.
+    </span>
+    <span *ngIf="date.errors.skyFuzzyDate?.yearRequired">
+      Please select a year.
+    </span>
+    <span *ngIf="date.errors.skyFuzzyDate?.futureDisabled">
+      Date cannot be in the future.
+    </span>
+    <span *ngIf="date.errors.skyFuzzyDate?.maxDate">
+      The current date is beyond the maximum allowed date.
+    </span>
+    <span *ngIf="date.errors.skyFuzzyDate?.minDate">
+      The current date is before the minimum allowed date.
+    </span>
+  </p>
+</div>
+
+<p>
+  Selected fuzzy date is {{ selectedDateForDisplay }}
+</p>
+
+<h2
+  class="sky-section-heading"
+>
+  Reactive form fuzzy datepicker
+</h2>
+
+<form
+  novalidate
+  style="max-width: 50%"
+  [formGroup]="reactiveForm"
+>
+  <sky-datepicker>
+    <input
+      formControlName="selectedDate"
+      skyFuzzyDatepickerInput
+      [dateFormat]="dateFormat"
+      [maxDate]="maxDate"
+      [minDate]="minDate"
+      [skyFuzzyDatepickerInput]
+      [skyDatepickerNoValidate]="noValidate"
+      [startingDay]="startingDay"
+      [yearRequired]="yearRequired"
+    />
+  </sky-datepicker>
+</form>
+
+<p>
+  Reactive form selected fuzzy date is {{ reactiveFormSelectedDateForDisplay }}
+</p>
+
+<p *ngIf="reactiveDate.errors && (reactiveDate.dirty || reactiveDate.touched)"
+  class="sky-error-label"
+>
+  <span *ngIf="reactiveDate.errors.skyFuzzyDate?.invalid">
+    Please enter a valid date.
+  </span>
+  <span *ngIf="reactiveDate.errors.skyFuzzyDate?.yearRequired">
+    Please select a year.
+  </span>
+  <span *ngIf="reactiveDate.errors.skyFuzzyDate?.futureDisabled">
+    Date cannot be in the future.
+  </span>
+  <span *ngIf="reactiveDate.errors.skyFuzzyDate?.maxDate">
+    The current date is beyond the maximum allowed date.
+  </span>
+  <span *ngIf="reactiveDate.errors.skyFuzzyDate?.minDate">
+    The current date is before the minimum allowed date.
+  </span>
+  <span *ngIf="reactiveDate.errors.required">
+    Date is required. Enter a value.
+  </span>
+</p>
+
+<p>
+  <button
+    class="sky-btn sky-btn-secondary"
+    (click)="clearSelectedDates()"
+  >
+    Clear selected dates
+  </button>
+  <button
+    class="sky-btn"
+    type="button"
+    (click)="setMinMaxDates()"
+  >
+    Set min and max dates
+  </button>
+  <button
+    class="sky-btn"
+    type="button"
+    (click)="toggleYearRequired()"
+  >
+    Toggle year required
+  </button>
+  <button
+    class="sky-btn"
+    type="button"
+    (click)="togglefutureDisabled()"
+  >
+    Toggle cannot be in cannot be in future
```

Comment:
We've got duplicate "cannot be in" instances that make this label gibberish. ... But even if we fix that, this is a vague label. I'd change this to something like `Toggle ability to add future dates` or `Toggle future date restriction`. (That said, if we can have 2 versions of this label, like I asked about for the previous button, we could tighten this up and make it clearer: `Enable future dates` and `Disable future dates`.)

Reply:
Updated to toggle switches.